1. Create a venv
2. Create a requirements.next
3. install autogen-agentchat 

In [ ]:
pip install autogen-agentchat autogen-core autogen-ext tiktoken openai dotenv


##### First Autogen Agent

In [3]:
import os
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient
from dotenv import load_dotenv

load_dotenv()

api_key = os.getenv('OPENAI_API_KEY')


###### Connecting to Model

In [7]:
model_client = OpenAIChatCompletionClient(model='gpt-4o',api_key=api_key)

In [5]:
agent_1 = AssistantAgent(name='my_assistant', model_client=model_client)

In [11]:
result = await agent_1.run(task="Tell me about you something")
result

TaskResult(messages=[TextMessage(id='0827eefd-59e5-4a9b-ab7f-15c68e312a1a', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 13, 53, 34, 387245, tzinfo=datetime.timezone.utc), content='Tell me about you something', type='TextMessage'), TextMessage(id='0223da3a-86aa-4fc0-8e70-2564fd2b2f24', source='my_assistant', models_usage=RequestUsage(prompt_tokens=118, completion_tokens=76), metadata={}, created_at=datetime.datetime(2025, 7, 5, 13, 53, 36, 947653, tzinfo=datetime.timezone.utc), content="I'm here to assist you by answering questions, providing explanations, and helping with a variety of tasks. My capabilities include understanding natural language, gathering and synthesizing information, and offering support in different areas like education, technology, and more. My goal is to make tasks easier and provide helpful, accurate information. If you have any specific questions or need assistance, feel free to ask!", type='TextMessage')], stop_reason

In [22]:
import pprint
result2 = await agent_1.run(task='What was my last question?')
#pprint.pprint(result2)
result2

TaskResult(messages=[TextMessage(id='103edce0-667a-4a3f-a395-5c618a7c5372', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 13, 56, 51, 33022, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'), TextMessage(id='500301bd-5957-45a2-b09d-513f09cf3391', source='my_assistant', models_usage=RequestUsage(prompt_tokens=640, completion_tokens=24), metadata={}, created_at=datetime.datetime(2025, 7, 5, 13, 56, 51, 826588, tzinfo=datetime.timezone.utc), content='Your last question was, "What was my last question?" If you need any more help, feel free to ask!', type='TextMessage')], stop_reason=None)


##### TaskResult(messages=[TextMessage(id='103edce0-667a-4a3f-a395-5c618a7c5372', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 13, 56, 51, 33022, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'), TextMessage(id='500301bd-5957-45a2-b09d-513f09cf3391', source='my_assistant', models_usage=RequestUsage(prompt_tokens=640, completion_tokens=24), metadata={}, created_at=datetime.datetime(2025, 7, 5, 13, 56, 51, 826588, tzinfo=datetime.timezone.utc), content='Your last question was, "What was my last question?" If you need any more help, feel free to ask!', type='TextMessage')], stop_reason=None)